src/react-native/04-intro-to-jsx/intro-to-jsx.md

```diff
@@ -201,9 +201,215 @@ Next, we want to render a list of states name in our application:
+const content = (
+  <Text>
+    {a === b // Ternaries are expressions.
+      ? "a and b are equal"
+      : "a and b are different"}
+  </Text>
+)
+```
+
+If ternaries seem excessive for any particular case, you can write all your logic in a separate function and invoke it from within JSX.
+
+```tsx
+function makeResult() {
+  return a === b ? "a and b are equal" : "a and b are different"
+}
+
+const content = <Text>{makeResult()}</Text>
+```
+
+#### Using loops
+
+JSX does not support traditional loop statements like `for`, `while`, or `do...while` directly within JSX.
+
+The example below will **not** work:
+
+```tsx
+const names = ['Alfa', 'Bravo', 'Charlie'];
+
+// This does not work
+const content = (
+  <View>
+    {
+      // Control flow does not belong in JSX
+      for (let name of names) {
+        <Text>name</Text>
+      }
+    }
+  </View>
+)
+```
+
+The `Array.map()` function is the most common and idiomatic way to render lists in JSX.
+It’s especially useful for rendering arrays of data as xml elements.
+
+```tsx
+const names = ["Alfa", "Bravo", "Charlie"]
+
+// This will work
+const content = (
+  <View>
+    {names.map((name) => {
+      return <Text key={name}>{name}</Text>
+    })}
+  </View>
+)
+```
+
+That will produce the following xml:
+
+```xml
+<View>
+  <Text>Alfa</Text>
+  <Text>Bravo</Text>
+  <Text>Charlie</Text>
+</View>
+```
+
+There are lots of ways to iterate over arrays in JavaScript with functions like `Array.map`, `Array.filter`, and `Array.reduce`. These all work in JSX!
+
+##### <Text id="the-key-prop">The `key` prop</Text>
+
+Did you notice the `key` prop in the example above?
+
+When rendering a list of elements, React needs a way to uniquely identify each element. This helps React understand which items have changed, been added, or removed, which is crucial for efficient re-rendering.
+
+Each key should be a unique identifier among siblings. Keys should be stable (not change over time), predictable (generated in a predictable manner), and unique (no two elements in the list should have the same key).
+
+It’s often convenient to use IDs from your data as keys. For example, if our data had `id` properties for each name, then we could use those as the `key` prop, even if there were duplicate names in the array:
+
+```tsx
+const names = [
+  { id: "550e8400", name: "Alfa" },
+  { id: "f47ac10b", name: "Bravo" },
+  { id: "5a3c9dd9", name: "Alfa" },
+  { id: "3d3f6f4d", name: "Charlie" },
+  { id: "aab3fcba", name: "Delta" },
+]
+
+const content = (
+  <View>
+    {names.map(({ id, name }) => {
+      return <Text key={id}>{name}</Text>
+    })}
+  </View>
+)
+```
+
+That will produce the following XML:
+
+```xml
+<View>
+  <Text>Alfa</Text>
+  <Text>Bravo</Text>
+  <Text>Alfa</Text>
+  <Text>Charlie</Text>
+  <Text>Delta</Text>
+</View>
+```
+
+🔦 _During development, If you forget to provide a `key` prop for items in an array React Native will log the following error to the browser’s console:_
```

Comment:
I’m not sure what the deal is here with the emoji and italics, but I would remove both.

Also:

- `If` -> `if`
- `array React` -> `array, React`
- There’s no browser.